### PR TITLE
Resolved Unsigned-pd support issue

### PR DESCRIPTION
**Pull request by quic-cgettibo**

### CODE REVIEW

File: src/fastrpc_apps_user.c [hunk 1]
```diff
@@ -342,6 +342,12 @@ void set_thread_context(int domain) {
 
 int fastrpc_session_open(int domain, int *dev) {
   int device = -1;
+
+  if(IS_SESSION_OPEN_ALREADY(domain)) {
+	  *dev = hlist[domain].dev;
+	  return 0;
```
Comment (ekanshibu) [inline src/fastrpc_apps_user.c]: indentation does not look correct

File: src/fastrpc_apps_user.c [hunk 2]
```diff
@@ -350,8 +356,9 @@ int fastrpc_session_open(int domain, int *dev) {
   return AEE_EUNKNOWN;
 }
 
-int fastrpc_session_close(int dev) {
-  close(dev);
+int fastrpc_session_close(int domain, int dev) {
```
Comment (quic-ekangupt) [inline src/fastrpc_apps_user.c:359]: add space after fastrpc_session_close

File: src/fastrpc_apps_user.c [hunk 3]
```diff
@@ -350,8 +356,9 @@ int fastrpc_session_open(int domain, int *dev) {
...
+	if(hlist[domain].dev == -1)
+		close(dev);
```
Comment (ekanshibu) [inline src/fastrpc_apps_user.c]: use uniform indentation

File: src/fastrpc_cap.c [hunk 4]
```diff
@@ -125,7 +125,7 @@ int fastrpc_get_cap(uint32_t domain, uint32_t attributeID, uint32_t *capability)
 
 bail:
    if(dev != -1)
-        fastrpc_session_close(dev);
+        fastrpc_session_close(dom, dev);
```
Comment (ekanshibu) [inline src/fastrpc_cap.c]: use uniform indentation

File: src/fastrpc_apps_user.c [hunk 5]
```diff
@@ -342,6 +342,12 @@ void set_thread_context(int domain) {
 
 int fastrpc_session_open(int domain, int *dev) {
   int device = -1;
+
+  if(IS_SESSION_OPEN_ALREADY(domain)) {
```
Comment (ekanshibu) [inline src/fastrpc_apps_user.c]: use space for indentation instead of tab

Comment (ekanshibu) [reply src/fastrpc_apps_user.c]: also give a space between if and (

File: src/fastrpc_apps_user.c [hunk 6]
```diff
@@ -350,8 +356,9 @@ int fastrpc_session_open(int domain, int *dev) {
   return AEE_EUNKNOWN;
 }
 
-int fastrpc_session_close(int dev) {
-  close(dev);
+int fastrpc_session_close(int domain, int dev) {
+  if(hlist[domain].dev == -1)
```
Comment (ekanshibu) [inline src/fastrpc_apps_user.c]: use space for indentation and space after "if"

**Review by quic-ekangupt — APPROVED** [review]

**Pull request merged**